src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/ApplyChangesToWorkspaceContext.cs

```diff
@@ -31,8 +29,8 @@ internal class ApplyChangesToWorkspaceContext : OnceInitializedOnceDisposed, IAp
         private readonly ConfiguredProject _project;
         private readonly IProjectLogger _logger;
         private readonly ExportFactory<IWorkspaceContextHandler>[] _workspaceContextHandlerFactories;
-        private IWorkspaceProjectContext _context;
-        private ExportLifetimeContext<IWorkspaceContextHandler>[] _handlers;
+        private IWorkspaceProjectContext? _context;
+        private ExportLifetimeContext<IWorkspaceContextHandler>[] _handlers = Array.Empty<ExportLifetimeContext<IWorkspaceContextHandler>>();
```

Comment:
In some cases it is easier to use singleton empty collections rather than `null`. Doing so can avoid lots of null checks or `!`, especially when there's an implicit relationship between being initialised and non-null (as happens a lot in this repo).

Reply:
👍

Reply:
I agree in some cases, but in this case it would be a bug that this is null that is no longer caught.